azure-iot-device/azure/iot/device/common/callable_weak_method.py

```diff
@@ -63,13 +64,18 @@ def __init__(self, object, method_name):
         self.method_name = method_name
 
     def _get_method(self):
-        return getattr(self.object_weakref(), self.method_name)
+        return getattr(self.object_weakref(), self.method_name, None)
 
     def __call__(self, *args, **kwargs):
         return self._get_method()(*args, **kwargs)
 
     def __eq__(self, other):
         return self._get_method() == other
+        # if inspect.isfunction(other):
```

Comment:
># if inspect.isfunction(other): [](start = 8, length = 31)

I've been debating this in my head.  I think you should leave the getattr change and revert the other 2 changes.

Reply:
I agree, especially since the getattr is the only thing required to fix the issue.

---
In reply to: [427012869](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r427012869) [](ancestors = 427012869)